common/rpc/interceptor/namespace_validator.go

```diff
@@ -103,6 +111,59 @@ func (ni *NamespaceValidatorInterceptor) LengthValidationIntercept(
 	return handler(ctx, req)
 }
 
+func (ni *NamespaceValidatorInterceptor) setNamespaceIfNotPresent(
+	namespaceRegistry namespace.Registry,
+	req interface{},
+) error {
+	namespaceName := GetNamespace(namespaceRegistry, req)
+	if namespaceName != "" {
```

Comment:
Why not just always set the namespace from the one in the token? Then we can drop all the token stuff from the rest of this file (extractNamespace gets simpler, checkNamespaceMatch goes away, etc.). Also this function and setNamespace gets simpler too

Reply:
Agree that we can simplify the logic here. My proposal is we can extract namespace from token, then we set it to request if it is empty. If it is not empty we validate it is the same as from token, if they are different then we return error. This will make sure the namespace from request is always the same as it from token, so we can remove all other logic around token validation in this file.

Reply:
We have to make sure this overwrite happen before authorizer interceptor because auth is done against the namespace set in the request field.